app/src/App.tsx

```diff
@@ -1,47 +1,96 @@
+import { useInterval } from '@react-corekit/use-interval'
 import { configureStore } from '@reduxjs/toolkit'
 import React, { useEffect, useState } from 'react'
+import Modal from 'react-modal'
 import { Provider } from 'react-redux'
 import { ThemeProvider } from 'styled-components'
 import Web3Provider from 'web3-react'
 
+import { FETCH_RPC_INTERVAL } from '../src/common/constants'
+
 import 'react-datepicker/dist/react-datepicker.css'
 import 'sanitize.css'
-
-import { HeaderNoRouter } from './components/common/layout/header'
 import { Main } from './components/main'
+import { ContentWrapper, ModalTitle } from './components/modal/common_styled' // here
+import { ConnectionModalNavigation, SettingsModalWrapper } from './components/modal/modal_your_connection'
 import SettingsViewContainer from './components/settings/settings_view'
 import { ApolloProviderWrapper } from './contexts/Apollo'
 import { ConnectedBalance, ConnectedWeb3 } from './hooks'
 import balanceReducer from './store/reducer'
 import theme from './theme'
 import { GlobalStyle } from './theme/global_style'
 import connectors from './util/connectors'
-import { getInfuraUrl } from './util/networks'
+import { getInfuraUrl, networkIds, networks } from './util/networks'
 import { checkRpcStatus, getNetworkFromChain } from './util/tools'
 
 const store = configureStore({ reducer: balanceReducer })
 
-const App: React.FC = () => {
+const App: React.FC = (props: any) => {
+  const { ...restProps } = props
   const windowObj: any = window
 
   const ethereum = windowObj.ethereum
   const networkId = ethereum && ethereum.chainId
   const [status, setStatus] = useState(true)
   const network = getNetworkFromChain(networkId)
+  const [settingsView, setSettingsView] = useState(false)
+
+  useInterval(() => {
+    if (status == false) {
+      setTimeout(() => {
+        if (status == false) setSettingsView(true)
+      }, 2000)
+    }
+
+    if (status == false && getNetworkFromChain(ethereum.chainId) == 100) {
+      localStorage.setItem(
+        'rpcAddress',
+        JSON.stringify({
+          url: networks[networkIds.XDAI].url || networkId.MAINNET,
+          network: networkIds.XDAI || networkIds.MAINNET,
+          index: 0,
+        }),
+      )
+    }
+  }, FETCH_RPC_INTERVAL)
```

Comment:
looking at the code this part is little fishy to me possibly...are we checking if it's relay?

Reply:
> So its still not working for me...
> 1)In networks go into xDai fuckup both default urls in this case rpc.xdaichain.com
> ![Screenshot 2021-06-09 at 22 18 00](https://user-images.githubusercontent.com/33226956/121423432-8df38900-c970-11eb-8040-8ef3b54996a7.png)
> 2)go and switch network to mainnet
> 
> So one indicator to check which rpc you are checking is to go into networks and see what you are calling in your case I noticed while on mainnet you are checking xDai url...
> ![Screenshot 2021-06-09 at 22 19 22](https://user-images.githubusercontent.com/33226956/121423711-d9a63280-c970-11eb-8ea6-a601c4964fc4.png)

yeah its because of the way i have infuraURL set to both the URLs so if one fails it will fall back to the other but ive actually been trying to find a fix for this today, noticed some of the same issues you mentioned here. let's see how rest of today/tonight goes.

Reply:
also thanks for the networks tab tip. ill start using that too